### DISCUSSION TITLE
[IntelliSense] Underline detected Tailwind class names

### DISCUSSION THREAD

**Discussion by ShayanTheNerd** [tailwindlabs/tailwindcss#14627] [Ideas]
Tailwind CSS IntelliSense offers a better DX compared to the UnoCSS counterpart. However, when we're typing class names, the UnoCSS extension underlines known classes as soon as the name is complete; like this:

![image](https://github.com/user-attachments/assets/973326fc-3fab-4fd0-adcf-a57db16f9170)

I think it'd be a good, little, opt-in feature to also have in Tailwind CSS IntelliSense.

**Comment by Asukaaaaaa** [comment]
find a vscode extension named `Tailwind CSS Highlight` could do the work

**Reply by ackvf** [reply]
This extension is useless. It will highlight utter crap as long as it resembles a tailwind classname. It won't help you spot errors whatsoever. It doesn't seem to use your tailwind.config either.

**Comment by ackvf** [comment]
What I did on a project was to add UnoCSS extension and unocss config to .gitignore and I just mirrored the tailwind configuration in unoconfig. It works for some advanced usecases too, but if you have complex rules or overrides, the effort in translating that to UnoCSS config just isn't worth it.

tldr; UnoCSS extension is good out of the box with just empty unoconfig file. Manageable with simple tailwind configuration. With complex rules you won't get highlights for those.

**Comment by minkoonim** [comment]
yeop this underline feature of unocss is so nice to have...

**Comment by plcdnl** [comment]
Could be nice also to help to migrate old projects to tailwindcss

**Comment by rublev** [comment]
bump
